includes/Orders.php

```
@@ -204,12 +227,48 @@ public function order_item_product( $product, $item ) {
+	 * those changes should only exist in memory. If WC stock functions later
+	 * call save() on the clone, this filter blocks the write for price-related
+	 * meta keys.
+	 *
+	 * @param null|bool $check     Return non-null to short-circuit update_metadata().
+	 * @param int       $object_id The post (product) ID.
+	 * @param string    $meta_key  The meta key being written.
+	 * @param mixed     $meta_value The meta value.
+	 *
+	 * @return null|bool Null to allow the write, true to block it.
+	 */
+	public function protect_product_price_meta( $check, $object_id, $meta_key, $meta_value ) {
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$meta_value'. (undefined)

(UnusedFormalParameter)